packages/api-client/src/generic/GenericAPI.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {AxiosRequestConfig} from 'axios';
+
+import {HttpClient} from '../http';
+
+export class GenericAPI {
+  constructor(private readonly client: HttpClient) {}
+
+  /**
+   * Get Data from a given URL.
+   */
+  public async get<T = unknown>(url: string): Promise<T> {
```

Comment:
Don't we want something even more generic? Something like a `request` method that could do a GET/POST... and take a full `AxiosRequestConfig` object?

Reply:
I would instead add post, put, and delete calls when needed. I was thinking about taking in a config, but I didn't have the use case for it yet.

Reply:
ok, fair enough, let's let it evolve as we go 👌